Plugins/Flow.Launcher.Plugin.Explorer/Search/SearchManager.cs

```diff
@@ -105,19 +107,21 @@ when ActionKeywordMatch(query, Settings.ActionKeyword.IndexSearchActionKeyword)
                 await foreach (var search in searchResults.WithCancellation(token).ConfigureAwait(false))
                     results.Add(ResultManager.CreateResult(query, search));
             }
+            catch (OperationCanceledException)
+            {
+                return results.ToList();
+            }
+            catch (EngineNotAvailableException)
+            {
+                throw;
+            }
             catch (Exception e)
             {
-                if (e is OperationCanceledException)
-                    return results.ToList();
-
-                if (e is EngineNotAvailableException)
-                    throw;
-
                 throw new SearchException(engineName, e.Message, e);
             }
-            
+
             results.RemoveWhere(r => Settings.IndexSearchExcludedSubdirectoryPaths.Any(
-                excludedPath => r.SubTitle.StartsWith(excludedPath.Path, StringComparison.OrdinalIgnoreCase)));
+                excludedPath => IsSubPathOf(r.SubTitle, excludedPath.Path)));
```

Comment:
what's the issue with this check?

Reply:
a path string starts with another doesn't necessarily mean it's contains the other one (unless ending with '\' is guaranteed). like `c:\program files` and `c:\program`